### ISSUE TITLE
Blocks: Design Review

### ISSUE THREAD

**Issue by coreymckrill** [WordPress/wordcamp.org#62]
We now have four of our five blocks up and running on the test site (the Schedule block is still in the works). We still have several issues in our [milestone](https://github.com/WordPress/wordcamp.org/milestone/1) before we officially launch them, but most of them are under-the-hood type things. What we'd like to get from you is a review of the design, in light of the fact that we've deviated a bit here and there from what was in the [original mockups](https://make.wordpress.org/community/2018/11/07/wordcamp-blocks-round-1-designs/). Ideally, you could open issues for requested changes or other problems that you see, and we can figure out together if they need to be included in the milestone or not.

Here's a summary of the things I'm aware of that are different. The idea with the checkboxes next to each item is that once you've reviewed it and either given it a pass or created issues to address it, we can check it off.

## General/Multiple

- [x] While all of the blocks include a setting to choose how to display post content (Full/Excerpt/None), none of them have an option to modify the _length_ of the excerpt. Modifying the excerpt length in a consistent, error-free way turned out to be really challenging, especially on the JS side, and it didn't seem important enough to hold up the v1.
- [ ] We added an option to show a Read More link after excerpts. It's not automatic because some themes append a Read More link already, which would cause a double-link situation. Note that the link _is_ automatically added at the end when Full content is selected, though.
- [x] The settings for featured images are a bit different than they are for avatars, even though the two types of images fulfill similar roles in the block content. Any thoughts on this?
- [ ] The icons in the select dropdowns have a `background-color: #f3f3f4`, which looks odd to me (@iandunn) for things like the `groups` dashicon in the Organizers block. Is that there to match the static background of the "blank" Gravatar, so that they're all consistently sized boxes?
- [ ] The "show category" option isn't consistent in the mockups for [Schedule](https://wp.invisionapp.com/share/GYOS8YM283F#/screens/327824817) and [Sessions](https://wp.invisionapp.com/share/QWOUA278STJ#/screens/328372422). Should it be? If so, which one should we use?

## Sessions

- [x] Both taxonomies, i.e. Tracks and Categories, are selectable from the dropdown instead of Tracks being a series of checkboxes in the Inspector Panel.
- [x] We're not using speaker avatars as the images for sessions. Instead we're using featured images. There's some discussion about this on [the original Make/Community post](https://make.wordpress.org/community/2018/10/30/wordcamp-block-sessions/#comment-26195).

## Sponsors

- [x] We decided to allow the Sponsor Level taxonomy to be selected in the dropdown, similar to taxonomies in the other post types. Because of this, we decided it wasn't necessary to output sponsor level headings within the block content. If people want headings and sponsors organized by level, they can add multiple Sponsors blocks, each with a different level selected, separated by Heading blocks.

If you spot other deviations, feel free to add them here.

**Comment by melchoyce** [comment]
Hey, looks like I don't have permissions to check things off!

Otherwise, makes sense — will open up anything I see as a new issue.

**Comment by melchoyce** [comment]
>  The settings for featured images are a bit different than they are for avatars, even though the two types of images fulfill similar roles in the block content. Any thoughts on this?

I imagined that the Sponsor images would always be the full-width of the container, and thus wouldn't need the image settings. But, maybe it makes sense to make it all the same and reuse the image block settings, in case people did want to reduce the size of the sponsor logos and float them inline. What do y'all think?

**Comment by coreymckrill** [comment]
> maybe it makes sense to make it all the same and reuse the image block settings, in case people did want to reduce the size of the sponsor logos and float them inline.

Yeah, I can see a scenario where you'd want to float the logo to the side and put the sponsor description next to it. My vote would be to make the image controls consistent with the other blocks.

**Comment by melchoyce** [comment]
> We added an option to show a Read More link after excerpts. It's not automatic because some themes append a Read More link already, which would cause a double-link situation.

Do you know which of our camp themes do this?

> Note that the link is automatically added at the end when Full content is selected, though.

I almost feel like this is the opposite of what we need — you probably don't need to read more if you've already read the full content.

**Comment by melchoyce** [comment]
> The icons in the select dropdowns have a background-color: #f3f3f4, which looks odd to me (@iandunn) for things like the groups dashicon in the Organizers block. Is that there to match the static background of the "blank" Gravatar, so that they're all consistently sized boxes?

Yes re: consistent sizing, but also, the icons were only there to help differentiate taxonomies from individual results. So, for example, individual session results should not show icons.

**Comment by melchoyce** [comment]
> The "show category" option isn't consistent in the mockups for Schedule and Sessions. Should it be? If so, which one should we use?

Let's go with the "Sessions" pattern, and do:

**Categories**
Show or hide session categories.

**Days**
Choose specific days to display.

**Tracks**
Choose specific tracks to display.

**Comment by coreymckrill** [comment]
@melchoyce 

> Do you know which of our camp themes (add a Read More link)?

Themes that add their own Read More link:

* Twenty Ten
* Twenty Eleven
* Twenty Thirteen
* Twenty Fourteen
* Twenty Fifteen
* Twenty Sixteen
* Twenty Seventeen
* WordCamp Base
* WordCamp Base Redux

Themes that don't add a Read More link:

* CampSite 2017
* Twenty Twelve
* Twenty Nineteen

> I almost feel like (adding a post link to the end of Full content) is the opposite of what we need — you probably don't need to read more if you've already read the full content.

I tend to agree. This was added based on a discussion somewhere where some people felt like linking the title and image were adequate, and that there needed to be a clear link to the individual post. I'll see if I can find that...

**Comment by melchoyce** [comment]
I wonder if we should update CampSite to display Read More and then just... rely on the themes. What do you think?

**Comment by coreymckrill** [comment]
@melchoyce The idea of eliminating the block option entirely and just letting the themes handle it is definitely appealing. I think it would be easy to do that with CampSite 2017 since it has no styles by default. But I'm not sure about Twenty Twelve / Twenty Nineteen. What do you think would be a workable solution for those?

**Comment by melchoyce** [comment]
Rely on the title being linked 🤷‍♀️

**Comment by ryelle** [comment]
Looking over this issue, it seems like _almost everything_ is sorted 🎉  

The remaining things I'm not sure about

- The organizers group icon seems to have disappeared a few iterations ago, did we want to bring that back?
- Now that speaker & organizer avatars are 50px (#177), should the group icons also be 50px?
- What was the outcome of the read more discussion above? Leave as it is currently?

@melchoyce @coreymckrill @iandunn Thoughts?

**Comment by melchoyce** [comment]
> The organizers group icon seems to have disappeared a few iterations ago, did we want to bring that back?

If the other groups still have icons, we should bring it back just to be consistent between all the blocks.

> Now that speaker & organizer avatars are 50px (#177), should the group icons also be 50px?

Can you add a screenshot of what it currently looks like? 

> What was the outcome of the read more discussion above? Leave as it is currently?

Sorry, short memory — how does it currently work, again?

**Comment by ryelle** [comment]
Larger avatars + smaller group icons:

![Screen Shot 2019-07-29 at 10 55 13 AM](https://user-images.githubusercontent.com/541093/62058422-6e756c00-b1ef-11e9-92c9-f80a35f26dba.png)

Larger avatars + no group icons (organizers, which you've said we should bring back, but just for reference):

![Screen Shot 2019-07-29 at 10 56 07 AM](https://user-images.githubusercontent.com/541093/62058457-7f25e200-b1ef-11e9-9873-5d7ec5856b0f.png)

**Comment by melchoyce** [comment]
> Larger avatars + smaller group icons:

Let's keep the group icons small. Emphasizes people over categories, and makes the group listings  more compact.

**Comment by ryelle** [comment]
> ryelle: What was the outcome of the read more discussion above? Leave as it is currently?
> melchoyce: how does it currently work, again?

The discussion above was about having a setting to show/hide the permalink to the post type's page (Speaker, Session, etc). It sounds like we removed the setting: the permalink is explicitly added to items when the full content is shown, but it's left up to the theme when excerpt is shown, since most themes inject a "Continue Reading" link on excerpts (but only when the content is cut off).

There's a point from @melchoyce that "I almost feel like this is the opposite of what we need — you probably don't need to read more if you've already read the full content." – which I agree with, we're actually hiding this link on [the WCUS speaker block.](https://2019.us.wordcamp.org/2019/08/01/announcing-our-speakers-round-4/)

So now I think the question is whether the full content should have the link at all.

**Comment by coreymckrill** [comment]
The permalink was added to the bottom of the full content because of concerns (from @melchoyce I think, but it's buried in Slack somewhere) that linking the post title was not obvious enough as a way to get to each single post in the post list.

I personally think linking the post title is sufficient, so if we can all agree on that now, I'd say let's go ahead and remove the link.

**Comment by ryelle** [comment]
Everything listed here has now been addressed 🎉  Closing out this issue, any other block UI/design bugs should be opened as new issues.

**Comment by iandunn** [comment]
🎉 Great work!
